mtcpclient/groupofconnections.go

```diff
@@ -29,23 +48,89 @@ func (gc GroupOfConnections) String() string {
 		nTotal, nDialing, nEstablished, nClosed, nError, nNotInitiated)
 }
 
-func (gc GroupOfConnections) isIn(status tcpclient.ConnectionStatus) bool {
-	for _, item := range gc {
+func (gc GroupOfConnections) containsAConnectionWithStatus(status tcpclient.ConnectionStatus) bool {
```

Comment:
are you using this? I see you are iterating over gc.connections in the following functions... I'd consider using this again; if we ever replace this slice by something different, we won't need to change N iterations..

Reply:
good catch, improved